src/services/analytics-service.ts

```diff
@@ -39,50 +39,61 @@ class AnalyticsTracking {
 
   private setReporting() {
     // don't send analytics other than in production deployment
+    // https://developers.google.com/analytics/devguides/collection/gtagjs/user-opt-out
     if (!config.shouldReportAnalytics) {
-      this.googleAnalytics.ga('set', 'sendHitTask', null);
+      (window as any)[`ga-disable-${googleAnalyticsKey}`] = true;
     }
   }
 
   // Track a pageview
   public trackPageView(pagePath: string) {
-    this.googleAnalytics.pageview(pagePath);
+    this.googleAnalytics.pageview({
+      page_path: pagePath
+    });
   }
 
   // Track an event
-  public trackEvent(ga: AnalyticsOptions) {
-    ga.species && this.setSpeciesDimension(ga.species);
-    ga.app && this.setAppDimension(ga.app);
-    ga.feature && this.setFeatureDimension(ga.feature);
+  public trackEvent(gtag: AnalyticsOptions) {
```

Comment:
both the `gtag` and the previously used `ga` are a strange name for the parameter of this function. I suggest we call it either `params`, or `options`, or `event`, or something similar.